### DISCUSSION TITLE
Does OXC pass all TC39 tests?

### DISCUSSION THREAD

**Discussion by DavidHancu** [oxc-project/oxc#4354] [Q&A] [answered]
As the title says, does it pass all tests from [tc39/test262](https://github.com/tc39/test262)?

**Comment by Dunqing** [comment]
The oxc parser passed 45859 tests and still has 4 negative tests failed. You can check it on our [snapshot](https://github.com/oxc-project/oxc/blob/test-visit/tasks/coverage/parser_test262.snap#L5)

**Comment by Boshen** [comment, marked as answer]
https://github.com/oxc-project/oxc/blob/test-visit/tasks/coverage/parser_test262.snap

```
parser_test262 Summary:

AST Parsed     : 45859/45859 (100.00%)
Positive Passed: 45859/45859 (100.00%)
Negative Passed: 3925/3929 (99.90%)
Expect Syntax Error: "language/import/import-assertions/json-invalid.js"
Expect Syntax Error: "language/import/import-assertions/json-named-bindings.js"
Expect Syntax Error: "language/import/import-attributes/json-invalid.js"
Expect Syntax Error: "language/import/import-attributes/json-named-bindings.js"
```

We can parse all test262 tests. The 4 "Expect Syntax Error" requires module resolution so is not done yet.

---

https://github.com/oxc-project/oxc/blob/test-visit/tasks/coverage/codegen_runtime_test262.snap

This shows our passed runtime tests, all failing cases are due to misconfigured environment.

---

@DavidHancu What do you need this information for?

**Reply by DavidHancu** [reply]
I'm trying to decide whether to use SWC or OXC for my project.

**Reply by Boshen** [reply]
Cool, feel free to ask more questions in our discord https://discord.gg/9uXCAwqQZW
